app/src/main/java/com/dessalines/rankmyfavs/ui/components/favlist/TierListScreen.kt

```diff
@@ -0,0 +1,317 @@
+
+        if (inputLimit.isBlank()) {
+            limit = null // remove the limit if empty
+        } else {
+            inputLimit.toIntOrNull()?.let {
+                if (it >= 0) {
+                    limit = it
+                }
+            }
+        }
+    }
+
+    Scaffold(
+        topBar = {
+            SimpleTopAppBar(
+                text = stringResource(R.string.tier_list),
+                navController = navController,
+            )
+        },
+        content = { padding ->
+            Column(
+                modifier =
+                    Modifier
+                        .padding(padding)
+                        .imePadding(),
+            ) {
+                OutlinedTextField(
+                    label = { Text(stringResource(R.string.tier_list_limit_description)) },
+                    modifier =
+                        Modifier
+                            .fillMaxWidth()
+                            .padding(SMALL_PADDING),
+                    value = inputLimit,
+                    onValueChange = { newLimit ->
+                        inputLimit = newLimit
+                    },
+                    singleLine = true,
+                )
+
+                ScreenshotBox(screenshotState = screenshotState) {
+                    TierList(tierList)
+                }
+            }
+        },
+        floatingActionButton = {
+            BasicTooltipBox(
+                positionProvider = tooltipPosition,
+                state = rememberBasicTooltipState(isPersistent = false),
+                tooltip = {
+                    ToolTip(stringResource(R.string.save))
+                },
+            ) {
+                FloatingActionButton(
+                    onClick = {
+                        screenshotState.capture()
+                        screenshotState.bitmap?.let {
+                            exportPngLauncher.launch("${favList?.name}_tier_list.png")
+                        }
+                    },
+                    shape = CircleShape,
+                ) {
+                    Icon(
+                        imageVector = Icons.Outlined.Save,
+                        contentDescription = stringResource(R.string.save),
+                    )
+                }
+            }
+        },
+    )
+}
+
+@Composable
+fun TierList(tierList: Map<String, List<FavListItem>>) {
+    Column(
+        modifier = Modifier.padding(horizontal = SMALL_PADDING),
+        verticalArrangement = Arrangement.spacedBy(SMALL_PADDING),
+    ) {
+        tierList.forEach { (tier, items) ->
+            TierSection(tier, items)
+        }
+    }
+}
+
+@Composable
+fun TierSection(
+    tier: String,
+    items: List<FavListItem>,
+) {
+    val tierColors =
+        mapOf(
+            "S" to Color(0XFFFF7F7F),
+            "A" to Color(0XFFFFBF7F),
+            "B" to Color(0XFFFFDF7F),
+            "C" to Color(0XFFFFFF7F),
+            "D" to Color(0XFFBFFF7F),
```

Comment:
At first I was tempted to say that we should be using material colors, but these tier lists have a standardized color scheme, so this is good.